cpp/include/raft/matrix/init.cuh

```diff
@@ -72,6 +73,23 @@ void fill(raft::resources const& handle,
   linalg::map(handle, inout, raft::const_op{scalar});
 }
 
+/**
+ * @brief create an identity matrix
+ * @tparam math_t data-type upon which the math operation will be performed
+ * @tparam idx_t indexing type used for the output
+ * @tparam layout_t layout of the matrix data (must be row or col major)
+ * @param[in] handle: raft handle
+ * @param[out] out: output matrix
+ */
+template <typename math_t, typename idx_t, typename layout_t>
+void eye(const raft::resources& handle, raft::device_matrix_view<math_t, idx_t, layout_t> out)
```

Comment:
What do you think about having this in `raft/matrix/diagonal.cuh`? It seems to fit nicely there isnce we already have `set_diagonal()` in that file. I think `init.cuh` for the matrix namespace has become a bit confusing for users since most of our other API headers are more targeted- like I think we should consider having `raft/matrix./fill.cuh` instead of `init.cuh`.